### PR TITLE
devscripts: execute linter on pkg directory

### PR DESCRIPTION
**Pull request by eduardoveiga**

### CODE REVIEW

File: devscripts/lint-code [hunk 1]
```diff
@@ -45,6 +46,9 @@ go() {
         docker-compose \
             -f docker-compose.yml -f docker-compose.dev.yml \
             exec $container golangci-lint run ./... $EXTRA_ARGS
+	[ $pkg == 1 ] && echo "Running Golang linter on pkg"
+        [ $pkg == 1 ] && docker-compose -f docker-compose.yml -f docker-compose.dev.yml exec $container sh -c "(cd ../pkg; golangci-lint run )"; $pkg = 0
+
```
Comment (otavio) [inline devscripts/lint-code:51]: Why are you using a variable to do this?

Comment (eduardoveiga) [reply devscripts/lint-code:51]: I want this linter run only one time, so I created this $pkg var 
this is the first idea I have but certanly there's a better way

Comment (otavio) [reply devscripts/lint-code:51]: @eduardoveiga I believe the below diff could work:
```diff
diff --git a/devscripts/lint-code b/devscripts/lint-code
index 8c06c3d..e9946ef 100755
--- a/devscripts/lint-code
+++ b/devscripts/lint-code
@@ -20,7 +20,7 @@ go() {
 
     [ ! -e .golangci.yaml ] && echo "ERROR: Run the script from the project root." && exit 1
 
-    for container in agent:docker agent:native api ssh cli; do
+    for container in agent:docker agent:native api ssh cli pkg; do
         echo "Running Golang linter on $container"
 
         [ "$fix" = "true" ] && EXTRA_ARGS="--fix" || EXTRA_ARGS=""
@@ -39,6 +39,12 @@ go() {
 
                 container=$(echo $container | cut -d':' -f 1)
                 ;;
+            pkg)
+                EXTRA_ARGS="$EXTRA_ARGS ../pkg/..."
+
+                # We use the API container to check the global pkg
+                container="api"
+                ;;
         esac
 
         # Run golangci-lint on a running container
```

Comment (eduardoveiga) [reply devscripts/lint-code:51]: `Running Golang linter on pkg
ERRO [linters context] typechecking error: pattern ../pkg/...: directory /go/src/github.com/shellhub-io/shellhub/pkg is outside module root (/go/src/github.com/shellhub-io/shellhub/api) 
WARN [runner] Can't run linter goanalysis_metalinter: durationcheck: failed prerequisites: [inspect@../pkg/...: analysis skipped: errors in package: [-: pattern ../pkg/...: directory /go/src/github.com/shellhub-io/shellhub/pkg is outside module root (/go/src/github.com/shellhub-io/shellhub/api)]] 
WARN [runner] Can't run linter makezero: makezero: analysis skipped: errors in package: [-: pattern ../pkg/...: directory /go/src/github.com/shellhub-io/shellhub/pkg is outside module root (/go/src/github.com/shellhub-io/shellhub/api)] 
WARN [runner] Can't run linter unused: buildir: analysis skipped: errors in package: [-: pattern ../pkg/...: directory /go/src/github.com/shellhub-io/shellhub/pkg is outside module root (/go/src/github.com/shellhub-io/shellhub/api)] 
ERRO Running error: buildir: analysis skipped: errors in package: [-: pattern ../pkg/...: directory /go/src/github.com/shellhub-io/shellhub/pkg is outside module root (/go/src/github.com/shellhub-io/shellhub/api)] `

there's not possible  
the ideal it's run the linter directly in pkg directory
so I used sh -c (cd ../pkg...

Comment (otavio) [reply devscripts/lint-code:51]: Please check if below does:
```diff
diff --git a/devscripts/lint-code b/devscripts/lint-code
index 8c06c3d..1e53542 100755
--- a/devscripts/lint-code
+++ b/devscripts/lint-code
@@ -20,7 +20,7 @@ go() {
 
     [ ! -e .golangci.yaml ] && echo "ERROR: Run the script from the project root." && exit 1
 
-    for container in agent:docker agent:native api ssh cli; do
+    for container in agent:docker agent:native api ssh cli pkg; do
         echo "Running Golang linter on $container"
 
         [ "$fix" = "true" ] && EXTRA_ARGS="--fix" || EXTRA_ARGS=""
@@ -39,6 +39,13 @@ go() {
 
                 container=$(echo $container | cut -d':' -f 1)
                 ;;
+            pkg)
+                # We have to move to pkg to allow to run the lint
+                EXTRA_ARGS="$EXTRA_ARGS ; (cd ../pkg; golangci-lint run ./...)"
+
+                # We use the API container to check the global pkg
+                container="api"
+                ;;
         esac
 
         # Run golangci-lint on a running container
```

Comment (eduardoveiga) [reply devscripts/lint-code:51]: `Running Golang linter on agent:docker
Running Golang linter on agent:native
Running Golang linter on api
Running Golang linter on ssh
Running Golang linter on cli
Running Golang linter on pkg
ERRO [linters context] typechecking error: stat /go/src/github.com/shellhub-io/shellhub/api/;: directory not found 
ERRO [linters context] typechecking error: stat /go/src/github.com/shellhub-io/shellhub/api/(cd: directory not found 
ERRO [linters context] typechecking error: stat /go/src/github.com/shellhub-io/shellhub/pkg;: directory not found 
ERRO [linters context] typechecking error: stat /go/src/github.com/shellhub-io/shellhub/api/golangci-lint: directory not found 
ERRO [linters context] typechecking error: stat /go/src/github.com/shellhub-io/shellhub/api/run: directory not found 
WARN [runner] Can't run linter goanalysis_metalinter: durationcheck: failed prerequisites: [inspect@./;: analysis skipped: errors in package: [-: stat /go/src/github.com/shellhub-io/shellhub/api/;: directory not found]] 
WARN [runner] Can't run linter makezero: makezero: analysis skipped: errors in package: [-: stat /go/src/github.com/shellhub-io/shellhub/api/;: directory not found] 
WARN [runner] Can't run linter unused: buildir: analysis skipped: errors in package: [-: stat /go/src/github.com/shellhub-io/shellhub/api/;: directory not found] 
ERRO Running error: buildir: analysis skipped: errors in package: [-: stat /go/src/github.com/shellhub-io/shellhub/api/;: directory not found] 
`

I don't understand this error "directory not found".  inside the containers. Looks like the ";" is considered part of the path

Comment (eduardoveiga) [reply devscripts/lint-code:51]: This way there's no new declared variable. I just put the command outside the case
`From 835298f6dcfe26619b14ccf6f505b1c85c7a5bcd Mon Sep 17 00:00:00 2001
From: Eduardo Kluwe Veiga <eduardoveiga@protonmail.ch>
Date: Mon, 21 Jun 2021 17:52:34 -0300
Subject: [PATCH] devscripts: execute linter on pkg directory

---
 devscripts/lint-code | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/devscripts/lint-code b/devscripts/lint-code
index 8c06c3d..a72ad74 100755
--- a/devscripts/lint-code
+++ b/devscripts/lint-code
@@ -17,6 +17,7 @@ EOF
 
 go() {
     local fix=$1
+    local pkg=1
 
     [ ! -e .golangci.yaml ] && echo "ERROR: Run the script from the project root." && exit 1
 
@@ -44,8 +45,10 @@ go() {
         # Run golangci-lint on a running container
         docker-compose \
             -f docker-compose.yml -f docker-compose.dev.yml \
-            exec $container golangci-lint run ./... $EXTRA_ARGS
+            exec $container golangci-lint run ./... $EXTRA_ARGS	
     done
+    echo "Running Golang linter on pkg"
+    docker-compose -f docker-compose.yml -f docker-compose.dev.yml exec $container sh -c "(cd ../pkg; golangci-lint run )"
 }
 
 vue() {
-- 
2.31.1`

Comment (otavio) [reply devscripts/lint-code:51]: It makes sense. Prepare the formal patch for review.

**Review by otavio — APPROVED** [review]

**Pull request merged**